homeassistant/components/homekit/type_media_players.py

```diff
@@ -68,17 +73,22 @@ def set_state(self, value):
         """Move switch state to value if call came from HomeKit."""
         _LOGGER.debug('%s: Set switch state to %s', self.entity_id, value)
         self.flag_target_state = True
-        service = self.on_service if value else self.off_service
+        service = self.state_service_map.on_service if value \
+            else self.state_service_map.off_service
         params = {ATTR_ENTITY_ID: self.entity_id}
         if self.mode == TOGGLE_MUTE:
             params.update({ATTR_MEDIA_VOLUME_MUTED: value})
         self.hass.services.call(DOMAIN, service, params)
 
     def update_state(self, new_state):
         """Update switch state after state changed."""
-        current_state = new_state.state if self.mode != TOGGLE_MUTE else \
-            new_state.attributes.get(ATTR_MEDIA_VOLUME_MUTED)
-        hk_state = current_state == self.on_state
+        current_state = new_state.state
+        if self.mode == TOGGLE_MUTE:
+            current_state = new_state.attributes.get(ATTR_MEDIA_VOLUME_MUTED)
+        hk_state = current_state == self.state_service_map.on_state
+        if self.mode == ON_OFF:
+            hk_state = current_state not in [self.state_service_map.off_state,
+                                             STATE_UNKNOWN, str(None)]
```

Comment:
A tuple would be better here: `(self.state_service_map.off_state, STATE_UNKNOWN, 'None')`.
I'm not sure though if `'None'` is really necessary here, but wouldn't be a big deal to leave it in either.